discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,136 @@
+import json
+import copy
+import logging  # pylint: disable=C0302
+from jsonschema import Draft7Validator, ValidationError
+logger = logging.getLogger(__name__)
+
+# https://app.quicktype.io/ -- JSON schema generator
+class ModelValidator:
+    '''
+        {key: value} :
+        {
+            model (first letter uppercase): {
+                model_schema: schema,
+                field_schema: {
+                    title: schema,
+                    mood: schema,
+                    ...
+            },
+                fields: [title, mood, ...]
+            }
+        }
+    '''
+    models_to_schema_and_fields_dict = {}
+
+    @classmethod
+    def validate(cls, to_validate, model, field=""):
```

Comment:
just a question - if this is a json field with sub validations like track_segments, the nested properties get validated here because those fields are marked as "required" correct?

Reply:
yes! when trying to insert a track with track_segments of an invalid structure, the validation logs the errors and overrides the value with a default:
```
[2020-06-01 21:22:24,116] {src.model_validator:34} (WARNING) - Error with Track instance {'track_segments': [{'a': 'b'}]}: 'duration' is a required property
[2020-06-01 21:22:24,116] {src.model_validator:34} (WARNING) - Error with Track instance {'track_segments': [{'a': 'b'}]}: 'multihash' is a required property
[2020-06-01 21:22:24,117] {src.model_validator:34} (WARNING) - Error with Track instance {'track_segments': [{'a': 'b'}]}: Additional properties are not allowed ('a' was unexpected)
[2020-06-01 21:22:24,117] {src.models:189} (WARNING) - Error: Instance {'track_segments': [{'a': 'b'}]} is not proper
Setting the default value None for field track_segments of type array
```